### PR TITLE
[#5213] allow user to select fleet in settings

### LINKED ISSUE #5213: Add way to select different fleets for use with the app
### User Story

As a sysadmin(devops is a buzzword) I want the app to be able to switch between different `eth` fleets without having to re-build the whole application. This is necessary to test infrastructure changes and `statusd` version changes on a __staging__ fleet before it's deployed to the fleet used in the public release.

### Description

Right now we have 3 fleets:
* [`eth.test`](https://consul.status.im/ui/do-ams3/nodes?s=eth.test) - Used by me for developing infra.
* [`eth.staging`](https://consul.status.im/ui/do-ams3/nodes?s=eth.staging) - Intended for `statusd` CI and testing new versions.
* [`eth.beta`](https://consul.status.im/ui/do-ams3/nodes?s=eth.beta) - The fleet currently used in the public releases.

Onle one of them is used in the app currently, the `eth.beta`.

#### Expected behavior
In order to test for example a new `statusd` version on `eth.staging` before deploying it to `eth.beta` we need to be able to select one of those in the App before logging in. This should be done in such a way that the node does not use any of the cached nodes.

#### Actual behavior
Currently when you log in you default to using the `boot`, `whisper` and `mail` nodes defined in two files:
* [`status-im/status-react/blob/develop/src/status_im/constants.cljs`](https://github.com/status-im/status-react/blob/develop/src/status_im/constants.cljs#L88)
* [`status-im/status-go/blob/develop/params/cluster.go`](https://github.com/status-im/status-go/blob/develop/params/cluster.go#L37)

This is hard to maintain and far too static.

### Additional Information
There was an issue cut for the `status-go` repo but it might be superfluous: https://github.com/status-im/status-go/issues/1100

I'm not sure what is the best way to do this but my gut is telling me that the best way would be to have a JSON file that `status-react` loads, that can be edited by devs/testers by hand if need be, that is then used to inform `status-go` library used by `status-react` on what to connect to. This way configuration will be just in one palce(`status-react`) and will be editable without the need to rebuild anything.

Something like:
```json
{
   "mainnet": {
       "default": "eth.beta",
       "eth.beta": {
           "boot": {
                "boot-01.do-ams3.eth.beta": "enode://ABCxyz@174.138.105.243:30404",
                "boot-02.do-ams3.eth.beta": "enode://XYZabc@206.189.243.57:30404",
                ...
           },
           "whisper": {
                ...
           },
           "mail": {
                ...
           }
       },
       "eth.staging": {
          ...
       }
   },
   "ropsten": {
      ...
   }
}
```

### PR DESCRIPTION
**Pull request by yenda**
fixes #5213 
(based off another PR for 5527)

### Summary:

- get wnodes from resources/config/fleets.json which is taken from fleets.status.im
- store wnodes by fleet and not by network since they are always the same
- reset wnodes settings during migration
- add option in developper menu to select fleet
- mailservers are now presented by their real name

### Steps to test:
- Open Status
- Check that fleet can be changed
- Check that mailserver settings are consistent when changing fleet (should randomly select a new mailserver)
(use cases list is wip)
- Check that go loglevel can be changed as well since the PR was merged into this one as well

status: ready

### CODE REVIEW

**Comment by adambabik** [comment]
> get wnodes from resources/config/fleets.json which is taken from fleets.status.im

Are they used as `StaticNodes` in `ClusterConfig`? We should pick up 1-2 randomly, not all of them. They are not counted in the peers discovery mechanism and we may end up with a lot of peers which will increase traffic.

**Comment by yenda** [comment]
@adambabik only one mail server node is connted when user logs in, picked randomly unless he picks one himself. They are separated from whisper nodes in the fleet config

**Review by janherich — APPROVED** [review]

File: src/status_im/data_store/realm/schemas/base/v9/account.cljs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+(ns status-im.data-store.realm.schemas.base.v9.account
+  (:require [taoensso.timbre :as log]
+            [cognitect.transit :as transit]
+            [clojure.set :as set]
+            [status-im.data-store.realm.schemas.base.v4.account :as v8]))
```
Comment (cammellos) [inline src/status_im/data_store/realm/schemas/base/v9/account.cljs]: `v4` account :)

**Comment by pedropombeiro** [comment]
@yenda, we're not supposed to see the mailserver names, right? (should be `Status mailserver A`, etc)

image

**Comment by yenda** [comment]
@PombeirP I did that on purpose because I don't think it makes sense to give them artificial names, if the list changes how do we update this ?

**Comment by pedropombeiro** [comment]
I don't think there is a problem if the list changes, if we just store an index into the list and we validate the index.
Maybe we should ask other stakeholders if they are fine with the change. cc @chadyj @jakubgs @adambabik

File: src/status_im/data_store/realm/schemas/account/v9/core.cljs [hunk 2]
```diff
@@ -23,4 +23,4 @@
              dapp-permissions/schema])
 
 (defn migration [old-realm new-realm]
-  (log/debug "migrating v9 account database: " old-realm new-realm))
\ No newline at end of file
+  (log/debug "migrating v9 account database: " old-realm new-realm))
```
Comment (mandrigin) [inline src/status_im/data_store/realm/schemas/account/v9/core.cljs:26]: unnecessary change

File: src/status_im/node/events.cljs [hunk 3]
```diff
@@ -5,10 +5,10 @@
...
  :node/start
- (fn [config]
-   (status/start-node config config/fleet)))
+ (fn [[config fleet]]
+   (status/start-node config fleet)))
 
 (re-frame/reg-fx
  :node/stop
- (fn [config]
+ (fn []
```
Comment (mandrigin) [inline src/status_im/node/events.cljs:13]: 👍 

btw, can we just do `(re-frame/reg-fx :node/stop status/stop-node)`?

**Review by mandrigin — APPROVED** [review]
I wonder if we can do a UI component that receives a list of entries, a set of labels and the db path to display a generic switcher for options... So we don't duplicate styles and views.

**Comment by adambabik** [comment]
@yenda 
> @adambabik only one mail server node is connted when user logs in, picked randomly unless he picks one himself. They are separated from whisper nodes in the fleet config

Right, but I wonder what do we mean by "wnode" in status-react? Because we also do have static nodes in status-go and these are Whisper nodes, not Mail Servers. I want to be sure that we do not put all Whisper nodes as static nodes in the configuration passed to status-go.

**Comment by adambabik** [comment]
> Maybe we should ask other stakeholders if they are fine with the change. 

I am ok with the new naming. It does reveal some info but it was still easy to figure out to which cloud provider a given IP belongs. On the pros side, it's gonna be much easier for us to communicate as "Mailserver F" required us to look up which node it actually is.

**Comment by jakubgs** [comment]
I see no problem with revealing names of our servers.

1. It's already public at https://fleets.status.im/
2. As @adambabik said, anyone can find out what providers we use by looking up IP block ownership
3. It easier to talk about issues with the fleet this way

<blockquote></blockquote>

**Comment by annadanchenko** [comment]
@yenda there are 2 issues:
### Issue 1. geth.log is empty for default INFO level, app has Storage permission (checked on Android)
Steps:
1. Install app and give it Storage permissions
2. Launch app, create account
3. Check Downloads folder for geth.log. Expected that it will contain some log entries (with info level). Currently, it's empty

### Issue 2. Node is started with beta.fleet even if staging is selected in the UI
In TestFairy log I can see node settings and it's "Fleet":"eth.beta" instead of selected Staging fleet. See https://app.testfairy.com/projects/4803622-status/builds/8581345/sessions/4400614902/?accessToken=uERQnSn18Agn/ZQD1kIK5xEfb8A
Steps:
1. Install app
2. Create new user
3. Change Fleet to staging
4. Log in
5. Check TestFairy log. Currently it contains "Fleet":"eth.beta" in node config, I would expect staging there



<blockquote><div><strong><a href="https://app.testfairy.com/projects/4803622-status/builds/8581345/sessions/4400614902/?accessToken=uERQnSn18Agn/ZQD1kIK5xEfb8A">TestFairy: Status 0.9.28 (7704) @2018</a></strong></div></blockquote>

**Comment by chadyj** [comment]
The new names are more descriptive and give more context so 👍

**Comment by mandrigin** [comment]
@yenda @annadanchenko I've fixed Issue 2

**Comment by yenda** [comment]
@mandrigin thanks for the fix, regarding better components I agree that is something I'll be looking into in the future

**Comment by annadanchenko** [comment]
@yenda 

Issue 3 unfortunately, there is no geth.log created for default or for TRACE log levels. Previously it was empty geth.log and now there is no file. (used new Android build from https://github.com/status-im/status-react/pull/5750#issuecomment-419004638)

Note that in node config it's now "LogEnabled":true,"LogLevel":"TRACE" but for some reason file is not created. TF session: https://app.testfairy.com/projects/4803622-status/builds/8582839/sessions/4400731560/?accessToken=v-dxnYzo-0SGf/ffU4vVq3wQDTM
Android 6.0.1, real Samsung Galaxy S6

Steps:
1. Install app and give it Storage permissions
2. Create account and log in
3. check Downloads folder for geth.log (it should be there)
4. Change log level to TRACE and log in again
5. check Downloads folder for geth.log (it should be there)

<blockquote><div><strong><a href="https://app.testfairy.com/projects/4803622-status/builds/8582839/sessions/4400731560/?accessToken=v-dxnYzo-0SGf/ffU4vVq3wQDTM">TestFairy: Status 0.9.28 (7774) @2018</a></strong></div></blockquote>

**Comment by annadanchenko** [comment]
@yenda all 3 previous issues are fixed, however there is new one:
Issue 4: If set log to DISABLED in ui then can't login: Wrong password is shown (I've used Paste, so I know that password is right)

TF with log and video: https://app.testfairy.com/projects/4803622-status/builds/8582984/sessions/4400758032/?accessToken=RPimHBcnsLE1j-ymGeNF-UjCzD8

Steps:
1. Install app and give it Storage permissions
2. Create account and log in (copy/paste password, so it's saved in the clipboard)
3. check Downloads folder for geth.log (it should be there)
4. Change log level to DISABLED and try to log in again by typing or pasting the password

<blockquote><div><strong><a href="https://app.testfairy.com/projects/4803622-status/builds/8582984/sessions/4400758032/?accessToken=RPimHBcnsLE1j-ymGeNF-UjCzD8">TestFairy: Status 0.9.28 (7778) @2018</a></strong></div></blockquote>

**Comment by asemiankevich** [comment]
@yenda @PombeirP  can we fix missing `DISABLED` in warning message so user knows he plans to disable?

Expected: `Change log level to DISABLED` when i want to disable

image

**Comment by asemiankevich** [comment]
last issue i have outlined is fixed, thanks @yenda ! I have also checked release build https://ci.status.im/job/status-react/job/release/job/release%252Ftest-log-level-selection/1/console - looks good!

the only thing is that latest PR contains a fix for DISABLED option and release build that i tested - does not 
cc @mandrigin @annadanchenko 

<blockquote><div><strong><a href="https://ci.status.im/job/status-react/job/release/job/release%252Ftest-log-level-selection/1/console">status-react » release » release/test-log-level-selection #1 Console [Jenkins]</a></strong></div></blockquote>

**Comment by yenda** [comment]
@asemiankevich regarding release build I don't think it matters it was just to verify that no logs are created by default in releases

**Comment by asemiankevich** [comment]
this one i can confirm @yenda :)

**Comment by asemiankevich** [comment]
@yenda @PombeirP  Not sure why when i set Log level for options ERROR or WARN it is shown as INFO in geth log on device? I am using Huawei P10 Lite, Android 8

**Comment by yenda** [comment]
If you set log level to warn or error there will also be info logs. Also
the app always starts with the log level in the environment variable and
switches when you log in

**Comment by asemiankevich** [comment]
@yenda @PombeirP i am asking about the naming of parameter itself
image

it is shown as INFO but i was expecting WARN / ERROR (according to UI settings applied)

**Comment by asemiankevich** [comment]
[geth.log.txt](https://github.com/status-im/status-react/files/2361125/geth.log.txt)

@yenda here is a log

**Comment by asemiankevich** [comment]
okay it looks good to me . I think we can merge it.
